crates/analyzer-core/src/preprocessor.rs

```diff
@@ -572,7 +573,8 @@ mod test {
 		let mut pp = PreprocessorState::new(|path| FileId::new(&db, path.into()), |_| unreachable!());
 
 		let test_id = FileId::new(&db, "<test-code>.p4".into());
-		let input = Buffer::new(&db, s.into());
+		let lsp_file = LspFile::new(&s.to_string());
+		let input = Buffer::new(&db, lsp_file);
```

Comment:
Based on this usage, maybe it'd be better to present a unified API with a constructor for `Buffer` and hide the `LspFile` abstraction within it.